collections/_article/godotsharp-packages-net8.md

```diff
@@ -0,0 +1,68 @@
+---
+title: "Godot C# packages move to .NET 8"
+excerpt: "On moving to the latest .NET LTS release, and what it means for users."
+categories: ["progress-report"]
+author: Raul Santos
+image: /storage/blog/covers/progress-update-csharp-3.webp
+date: 2024-11-26 17:00:00
+---
+
+A new version of .NET is released every year on November, alternating Long Term Support (LTS) and Standard Term Support (STS) releases. The quality of all releases is the same. The only difference is the length of  support. LTS releases are supported for 3 years, while STS releases are supported for 1 year and a half.
+
+With the release of Godot 4.0 we moved our C# packages from targeting Mono to .NET 6. This version of .NET was released on November 8, 2021 and ended support on November 12, 2024.
+
+With .NET 6 ending support, Godot C# packages in 4.4 will now target the next LTS release .NET 8. Thanks to [RedworkDE](https://github.com/RedworkDE) and [Paul Joannon](https://github.com/paulloz) for the great work in [GH-92131](https://github.com/godotengine/godot/pull/92131).
+
+<figure>
+	<picture>
+		<source srcset="/storage/blog/dotnet/release-schedule-dark.svg" media="(prefers-color-scheme: dark)">
+		<img class="lightbox-ignore" style="background-color: transparent;" src="/storage/blog/dotnet/release-schedule-light.svg">
+	</picture>
+	<figcaption><small>.NET release cadence from the <a href="https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core">.NET official support policy</a>.</small></figcaption>
+</figure>
+
+## What does this mean for users?
+
+This means your project now needs to target .NET 8 or newer, otherwise it will be incompatible with the new `GodotSharp` packages. But you can still use libraries that target older .NET versions.
+
+Existing projects will be automatically upgraded to target `net8.0` when they are opened with Godot 4.4. If your project already targets .NET 8 or newer, nothing will change. Thanks to [Andreia Gaita](https://github.com/shana) for implementing this in [GH-82927](https://github.com/godotengine/godot/pull/82927).
+
+## What took so long?
+
+The version of .NET that a library targets becomes the minimum required version for consumers of that library. That is, if the `GodotSharp` package targets `net7.0`, your game project needs to target `net7.0` or newer. We didn't want to force our users to upgrade to a non-LTS release.
+
+When .NET 8 was released, it would've been a good time to bump the target version in Godot C# packages, since .NET 8 is an LTS release. Unfortunately, there were some breaking changes in .NET 7[^1] that prevented us from upgrading and making the required changes took us longer than we hoped.
```

Comment:
```suggestion
When .NET 8 was released in November 2023, it would've been a good time to bump the target version in Godot C# packages, since .NET 8 is an LTS release. Unfortunately, there were some breaking changes in .NET 7[^1] that prevented us from upgrading, and making the required changes took us longer than we hoped.
```